proposals/umoci.md

```diff
@@ -0,0 +1,192 @@
+This proposal for umoci's inclusion is intended to make it easier for users who do not wish to implement the OCI Image Specification themselves to have a basic building block they can use for whatever system they are developing.
+While umoci is usable on its own, it is expected that the vast majority of umoci's users will be consuming it as part of a larger build system which uses umoci internally to abstract away the specifics of the OCI Image Specification.
+
+> *What about image-tools?*
+
+The original intention of umoci was to be a wrapper around the OCI image-tools and slowly upstream the work.
+Unfortunately, in the past 4 years, the image-tools project has not developed at a consistent pace -- and umoci works today and is fairly widely used.
+It is far more important that the OCI provide a solid "works out of the box" image manipulation tool, and umoci is (in our view) the best candidate for this role.
+
+In a future proposal, we may decide to sub-tree merge the OCI image-tools project into umoci (and archive the original project) in order to maintain the legacy of the work done by previous contributors and maintainers.
+Alternatively we may decide to split out the validation code from the OCI image-tools project and place it inside an OCI conformance project.
+However, we believe that these decisions should be considered a separate topic to the question of umoci's inclusion in the OCI.
+
+> *Why bless umoci over other alternative tools?*
+
+umoci is the only OCI image manipulation tool which was designed to be usable as a component to build other image-building tools.
+This has proven to be a practical benefit, with several tools ([KIWI][kiwi], [stacker][stacker], and several others) being built on top of umoci.
+In addition, umoci has been used by the [LXC project][lxc-oci] to support OCI-based images from *outside* the OCI ecosystem, something that was only reasonable because of umoci's unopinionated stance on images.
+umoci has seen production usage for several years as part of the [Open Build Project][obs]'s build system, building container images using [kiwi][kiwi].
+
+[kiwi]: https://osinside.github.io/kiwi/building/build_docker_container.html
+[stacker]: https://github.com/anuvu/stacker
+[lxc-oci]: https://github.com/lxc/lxc/blob/lxc-4.0.2/templates/lxc-oci.in
+[obs]: https://openbuildservice.org/
+
+> *How do we avoid the runc issue with implementation-specific quirks becoming a de-facto standard?*
+
+When developing new features in umoci (which are within the scope of the OCI Image Specification), a strong effort will be made to include those features in the upstream specification.
+In addition, any new features that are in the scope of the OCI Image Specification will be marked as strictly experimental and unsupported, to ensure that users do not depend on their stability until the relevant feature (or an equivalent feature) is included in the OCI Image Specification.
```

Comment:
Yes. While we might ship some experimental OCIv2 implementation in an `umoci` release, it will be explicitly unsupported and only intended for real-world testing until whatever OCIv2 ends up looking like lands in the actual image-spec.

Reply:
And it should be noted that one of the things #86 does is to explicitly require this for all Reference Implementations.